scanner/src/test/kotlin/scanners/fossid/FossIdTest.kt

```diff
@@ -0,0 +1,829 @@
+import org.ossreviewtoolkit.clients.fossid.model.identification.identifiedFiles.IdentifiedFile
+import org.ossreviewtoolkit.clients.fossid.model.identification.ignored.IgnoredFile
+import org.ossreviewtoolkit.clients.fossid.model.identification.markedAsIdentified.License
+import org.ossreviewtoolkit.clients.fossid.model.identification.markedAsIdentified.LicenseFile
+import org.ossreviewtoolkit.clients.fossid.model.identification.markedAsIdentified.MarkedAsIdentifiedFile
+import org.ossreviewtoolkit.clients.fossid.model.status.DownloadStatus
+import org.ossreviewtoolkit.clients.fossid.model.status.ScanState
+import org.ossreviewtoolkit.clients.fossid.model.status.ScanStatus
+import org.ossreviewtoolkit.clients.fossid.runScan
+import org.ossreviewtoolkit.model.CopyrightFinding
+import org.ossreviewtoolkit.model.CuratedPackage
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.LicenseFinding
+import org.ossreviewtoolkit.model.OrtIssue
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.ScanSummary
+import org.ossreviewtoolkit.model.ScannerRun
+import org.ossreviewtoolkit.model.Severity
+import org.ossreviewtoolkit.model.TextLocation
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.VcsType
+import org.ossreviewtoolkit.model.config.DownloaderConfiguration
+import org.ossreviewtoolkit.model.config.ScannerConfiguration
+import org.ossreviewtoolkit.utils.test.shouldNotBeNull
+
+class FossIdTest : WordSpec({
+    beforeSpec {
+        mockkStatic("org.ossreviewtoolkit.clients.fossid.ExtensionsKt")
+    }
+
+    "version" should {
+        "contain the version extracted from the login page of the FossID server" {
+            val fossId = createFossId(createConfig())
+
+            fossId.version shouldBe FOSSID_VERSION
+        }
+
+        "be empty if no version could be extracted from the login page" {
+            val config = createConfig(loginPage = "This is an unexpected login page.")
+            val fossId = createFossId(config)
+
+            fossId.version shouldBe ""
+        }
+    }
+
+    "scanPackages" should {
+        "create a new scan for an existing project" {
+            val projectCode = projectCode(PROJECT)
+            val scanCode = scanCode(PROJECT, null)
+            val config = createConfig(deltaScans = false)
+            val vcsInfo = createVcsInfo()
+            val scan = createScan(vcsInfo.url, "${vcsInfo.revision}_other", scanCode)
+
+            val service = config.createService()
+                .expectProjectRequest(projectCode)
+                .expectListScans(projectCode, listOf(scan))
+                .expectCheckScanStatus(scanCode, ScanState.FINISHED)
+                .expectCreateScan(projectCode, scanCode, vcsInfo)
+                .expectDownload(scanCode)
+                .mockFiles(scanCode, identifiedRange = 1..2, markedRange = 1..2)
```

Comment:
Here you don't need some `identifiedRange` nor `marckedRange` since you don't test it,